Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs

```diff
@@ -470,9 +470,7 @@ public void Detach ()
 		/// </remarks>
 		public ThreadInfo ActiveThread {
 			get {
-				lock (slock) {
-					return activeThread;
-				}
+				return activeThread;
```

Comment:
activeThread is written to outside the lock in [OnTargetEvent](https://github.com/mono/debugger-libs/blob/3c2aeea23797414e86fa225710b217d2fe3aa50e/Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs#L1203-L1204)

Reply:
Yep, I saw that. Are you saying that this lock is needed? or that locking is needed in OnTargetEvent? or both?

Reply:
Oh, it seems like reference assignment is also atomic. Should be good 👍